packages/contracts/contracts/factory/DAOFactory.sol

```diff
@@ -2,155 +2,152 @@
-    event DAOCreated(string name, address indexed token, address indexed voting);
+    /// @notice Thrown if `PluginSettings` array is empty, and no plugin is provided.
+    error NoPluginProvided();
+
+    /// @notice Thrown if none of the provided `PluginSettings` is a governance plugin.
+    error NoGovernancePluginProvided();
 
     /// @notice The constructor setting the registry and token factory address and creating the base contracts for the factory to clone from.
     /// @param _registry The DAO registry to register the DAO by its name.
-    /// @param _tokenFactory The token factory for optional governance token creation.
-    constructor(DAORegistry _registry, TokenFactory _tokenFactory) {
+    /// @param _pluginSetupProcessor The addres of PluginSetupProcessor.
+    constructor(DAORegistry _registry, PluginSetupProcessor _pluginSetupProcessor) {
         daoRegistry = _registry;
-        tokenFactory = _tokenFactory;
+        pluginSetupProcessor = _pluginSetupProcessor;
 
-        setupBases();
+        daoBase = address(new DAO());
     }
 
-    /// @notice Creates a new DAO with the `ERC20Voting` component installed and deploys a new [ERC-20](https://eips.ethereum.org/EIPS/eip-20) governance token if the corresponding configuration is passed.
-    /// @param _daoConfig The name and metadata hash of the DAO.
-    /// @param _voteConfig The configuration used to set up the the majority voting.
-    /// @param _tokenConfig The configuration used to create a new token.
-    /// @param _mintConfig The configuration used to mint the newly created tokens.
-    /// @param _trustedForwarder The address of the trusted forwarder required for meta transactions.
-    function createERC20VotingDAO(
-        DAOConfig calldata _daoConfig,
-        VoteConfig calldata _voteConfig,
-        TokenFactory.TokenConfig calldata _tokenConfig,
-        TokenFactory.MintConfig calldata _mintConfig,
-        address _trustedForwarder
-    )
+    function createDao(DAOSettings calldata _daoSettings, PluginSettings[] calldata pluginSettings)
         external
-        returns (
-            DAO dao,
-            ERC20Voting voting,
-            ERC20VotesUpgradeable token,
-            MerkleMinter minter
-        )
+        returns (DAO dao)
     {
-        if (_mintConfig.receivers.length != _mintConfig.amounts.length)
-            revert MintArrayLengthMismatch({
-                receiversArrayLength: _mintConfig.receivers.length,
-                amountsArrayLength: _mintConfig.amounts.length
-            });
-
-        dao = createDAO(_daoConfig, _trustedForwarder);
-
-        // Create token and merkle minter
-        dao.grant(address(dao), address(tokenFactory), dao.ROOT_PERMISSION_ID());
-        (token, minter) = tokenFactory.createToken(dao, _tokenConfig, _mintConfig);
-        dao.revoke(address(dao), address(tokenFactory), dao.ROOT_PERMISSION_ID());
+        // Check if no plugin is provided.
+        if (pluginSettings.length == 0) {
```

Comment:
why do we revert if no plugin is provided ? sometimes people just want to create dao without any plugin (just a boilerplate).

Reply:
we will have an admin governance plugin, that a dao can start with, so that it will not be locked.
while true msg.sender can be the initialOwner, but end user will not have suitable UI to interact with dao.